### DISCUSSION TITLE
Does LDOS obey superposition?

### DISCUSSION THREAD

**Discussion by Andeloth** [NanoComp/meep#2249] [Q&A] [answered]
Lets say that I would like to know the LDoS at each point in a grid, independent of the spatial characteristics of any sources (lets say we assume a specific frequency spectrum). In other words, I would like to know the Density of States for my whole system and I was thinking of doing that by summing up the local density of states for each point in the system. Since meep calculates ldos **only** where a source is located, to me it seems I have 2 ways of calculating the density of states for my geometry:

1. Define a point dipole source at a point in the grid, run the simulation and calculate ldos. Repeat for all points on the grid, then integrate the results over the grid.
2. Define a single source over the grid (e.g. `Source(GaussianSource(...)), center=mp.Vector3(), size=grid_size)`), and calculate the ldos for the whole region.

I would naively like to believe that summing up the results of the first calculation should equal the second because of linear superposition. However, I feel like there is an error in line of thinking. If there is, could someone please point it out? (I just think there is no way it's that simple.)

If the two calculations give different results, what would be the physical meaning of both calculations?

If neither of those are proportional to the system's density of states, is there a way to calculate it in meep at the present moment?

**Comment by stevengj** [comment, marked as answer]
Nope, the second way won't work, because the LDOS is not a linear function of the fields.

Physically, the first calculation corresponds to exciting the fields with a spatially incoherent source (e.g. like spontaneous emission or thermal emission), whereas the second calculation corresponds to a spatially coherent source (like an antenna array).  They are totally inequivalent.
